### PR TITLE
Update k8s dependencies to v0.30.2

### PR DESCRIPTION
**Pull request by thunderboltsid**
This also updates some otel lib versions to fix a CVE.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
-	github.com/onsi/gomega v1.31.1
-	k8s.io/api v0.29.1
-	k8s.io/apimachinery v0.29.1
-	k8s.io/client-go v0.29.1
+	github.com/onsi/ginkgo/v2 v2.17.1
+	github.com/onsi/gomega v1.32.0
+	k8s.io/api v0.30.1
+	k8s.io/apimachinery v0.30.1
+	k8s.io/client-go v0.30.1
 	k8s.io/cloud-provider v0.29.1
```
Comment (JoelSpeed) [inline go.mod:15]: You probably want this version to align with the other k8s.io updates as well

Comment (thunderboltsid) [reply go.mod]: Good catch!

**Review by yannickstruyf3 — APPROVED** [review]

**Review by tuxtof — APPROVED** [review]
LGTM

**Pull request merged**